windows/ReactNativeCameraCPP/CameraRotationHelper.cpp

```diff
@@ -0,0 +1,242 @@
+        {
+            case SimpleOrientation::Rotated90DegreesCounterclockwise:
+                return 270;
+            case SimpleOrientation::Rotated180DegreesCounterclockwise:
+                return 180;
+            case SimpleOrientation::Rotated270DegreesCounterclockwise:
+                return 90;
+            case SimpleOrientation::NotRotated:
+            default:
+                return 0;
+        }
+    }
+
+    SimpleOrientation CameraRotationHelper::ConvertDisplayOrientationToSimpleOrientation(DisplayOrientations orientation)
+    {
+        SimpleOrientation result;
+        switch (orientation)
+        {
+        case DisplayOrientations::Landscape:
+            result = SimpleOrientation::NotRotated;
+            break;
+        case DisplayOrientations::PortraitFlipped:
+            result = SimpleOrientation::Rotated90DegreesCounterclockwise;
+            break;
+        case DisplayOrientations::LandscapeFlipped:
+            result = SimpleOrientation::Rotated180DegreesCounterclockwise;
+            break;
+        case DisplayOrientations::Portrait:
+        default:
+            result = SimpleOrientation::Rotated270DegreesCounterclockwise;
+            break;
+        }
+
+        // Above assumes landscape; offset is needed if native orientation is portrait
+        if (m_displayInformation.NativeOrientation() == DisplayOrientations::Portrait)
+        {
+            result = AddOrientations(result, SimpleOrientation::Rotated90DegreesCounterclockwise);
+        }
+
+        return result;
+    }
+
+    SimpleOrientation CameraRotationHelper::SubtractOrientations(SimpleOrientation a, SimpleOrientation b)
+    {
+        auto aRot = ConvertSimpleOrientationToClockwiseDegrees(a);
+        auto bRot = ConvertSimpleOrientationToClockwiseDegrees(b);
+        // Add 360 to ensure the modulus operator does not operate on a negative
+        auto result = (360 + (aRot - bRot)) % 360;
+        return ConvertClockwiseDegreesToSimpleOrientation(result);
+    }
+
+    SimpleOrientation CameraRotationHelper::MirrorOrientation(SimpleOrientation orientation)
+    {
+        // This only affects the 90 and 270 degree cases, because rotating 0 and 180 degrees is the same clockwise and counter-clockwise
+        switch (orientation)
+        {
+        case SimpleOrientation::Rotated90DegreesCounterclockwise:
+            return SimpleOrientation::Rotated270DegreesCounterclockwise;
+        case SimpleOrientation::Rotated270DegreesCounterclockwise:
+            return SimpleOrientation::Rotated90DegreesCounterclockwise;
+        }
+        return orientation;
+    }
+
+    SimpleOrientation CameraRotationHelper::AddOrientations(SimpleOrientation a, SimpleOrientation b)
+    {
+        auto aRot = ConvertSimpleOrientationToClockwiseDegrees(a);
+        auto bRot = ConvertSimpleOrientationToClockwiseDegrees(b);
+        auto result = (aRot + bRot) % 360;
+        return ConvertClockwiseDegreesToSimpleOrientation(result);
```

Comment:
can either of these be negative, if so do we need to +360 before doing %360?

Reply:
No, the value will be >= 0, either 0,90,180 or 270. The rotation helper code are exact copy from MSDN, except I converted it from CSharp to C++/WinRT.